lib/src/over_react_test/custom_matchers.dart

```diff
@@ -337,3 +339,137 @@ Matcher throwsPropError_Combination(String propName, String prop2Name, [String m
+///
+/// Related: [emitsLogs], [emitsNoLogs]
+Matcher emitsLog(String expected, {ConsoleConfiguration consoleConfig}) =>
+    _LoggingFunctionMatcher(anyElement(contains(expected)), config: consoleConfig);
+
+/// A Matcher used to compare a list of logs against a provided matcher.
+///
+/// __Examples:__
+///
+/// To look for a specific `String` in any log index, the best solution is to use
+/// [emitsLog], but the behavior can be mimicked by passing in the correct `Iterable`
+/// matchers.
+/// ```dart
+///   expect(callbackFunction, emitsLogs(anyElement(contains('I expect this log'))));
+/// ```
+///
+/// When passed a `List`, the matcher will do an equality check on the actual
+/// log `List`.
+///
+/// Alternatively, the `String` can be wrapped in a `contains` to check the
+/// if the comparable index contains that substring.
+/// ```dart
+///   expect(callbackFunction, emitsLogs(['I expect this log', 'And this Log']));
+///   expect(callbackFunction, emitsLogs([
+///     contains('I expect'),
+///     contains('And this'),
+///   ]));
+/// ```
+///
+/// All usual `Iterable` matchers can also be used.
+/// ```dart
+///   expect(callbackFunction, emitsLogs(containsAll(['I expect this log'])));
+///   expect(callbackFunction, emitsLogs(containsAllInOrder(['I expect this log'])));
+///   expect(callbackFunction, emitsLogs(hasLength(1)));
+/// ```
+///
+/// Related: [emitsLog], [emitsNoLogs]
+Matcher emitsLogs(dynamic expected, {ConsoleConfiguration consoleConfig}) =>
+    _LoggingFunctionMatcher(expected, config: consoleConfig);
+
+/// A matcher to verify that a callback function does not emit any logs.
+///
+/// Related: [emitsLogs]
+final Matcher emitsNoLogs = _LoggingFunctionMatcher(isEmpty);
+
+/// The string used to identify a `propType` error.
+const _propTypeErrorMessage = 'Failed prop type';
+
+/// A matcher used to assert an actual `List` contains expected `propType`
+/// warnings.
+///
+/// The actual value can be a callback function, which will result in the matcher
+/// asserting that the expected `propType` warnings appear during the runtime
+/// of the callback.
+///
+/// Related: [_LoggingFunctionMatcher]
+class _PropTypeLogMatcher extends _LoggingFunctionMatcher {
+  _PropTypeLogMatcher(expected)
+      : super(expected,
+      description: 'emits the propType warning',
+      name: 'propType warning');
+
+  final _filter = contains(_propTypeErrorMessage);
+
+  @override
+  featureValueOf(actual) {
+    if (actual is! Function() && actual is! List) {
+      throw ArgumentError('The actual value must be a callback or a List.');
+    }
+
+    var logs = actual is List ? actual : recordConsoleLogs(actual, errorConfig);
+    return logs.where((log) => _filter.matches(log, {})).toList();
+  }
+}
+
+/// Matcher used to check for a specific `propType` warning being emitted during
+/// the runtime of a callback function.
+///
+/// Has the same underlying logic as [emitsLog], with the difference being that
```

Comment:
Can we add notes in all these comments that any errors thrown by the function are ignored?

I know it's kind of redundant, but that information could be extremely valuable when debugging, and should be as obvious as possible.